### PR TITLE
Applied Shields v2 UX to settings and shields panel and use FP V2 by default

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/9194
Resolves https://github.com/brave/brave-browser/issues/9975

-------------------
Screen Shot 2020-05-29 at 12 05 59 PM

-------------------
Screen Shot 2020-05-29 at 1 15 26 PM

-------------------
Screen Shot 2020-05-29 at 1 15 34 PM

-------------------
Screen Shot 2020-05-29 at 1 15 43 PM

-------------------
Screen Shot 2020-05-28 at 8 01 15 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
@pes10k @rebron I think this PR is ready to review :)

**Comment by rebron** [comment]
Labels and renaming looks great. Will leave the rest to @pes10k.

**Comment by pes10k** [comment]
This all looks great to me, terrific @simonhong !

**Comment by simonhong** [comment]
@bridiver ping to owners review.

File: components/brave_shields/browser/brave_shields_util.cc [hunk 1]
```diff
@@ -106,6 +106,38 @@ ControlType ControlTypeFromString(const std::string& string) {
...
 }
 
+std::string FingerprintingControlTypeToString(ControlType type) {
+  switch (type) {
+    case ControlType::ALLOW:
+      return "allow";
+    case ControlType::BLOCK:
+      return "block";
+    case ControlType::BLOCK_THIRD_PARTY:
+      FALLTHROUGH;
```
Comment (bridiver) [inline components/brave_shields/browser/brave_shields_util.cc:116]: doesn't this also need a v2 check?

Comment (simonhong) [reply components/brave_shields/browser/brave_shields_util.cc:116]: In the UI, same string `default` is only used for `BLOCK_THIRD_PARTY` and `DEFAULT`.

**Comment by simonhong** [comment]
@rebron I'll revisit and update this PR again after fingerprint v2 is enabled by default.
@bridiver told me that these new option names are valid only for v2. cc: @pes10k

**Comment by pes10k** [comment]
Yep, these are for v2 fingerprinting, but we're aiming to ship v2 in 1.11, so should all line up :)

Right now the v2 fingerprinting protects *haven't* been fully implemented, but *have* eclipsed what was available in v1, so switching over to v2, even as is, is strictly superior

File: components/brave_shields/browser/brave_shields_util_unittest.cc [hunk 2]
```diff
@@ -701,90 +701,90 @@ TEST_F(BraveShieldsUtilTest, GetFingerprintingControlType_Default) {
...
   EXPECT_EQ(ControlType::BLOCK, setting);
 
-  /* BLOCK_THIRD_PARTY */
+  /* DEFAULT */
   map->SetContentSettingCustomScope(
       ContentSettingsPattern::Wildcard(), ContentSettingsPattern::Wildcard(),
-      ContentSettingsType::PLUGINS, brave_shields::kFingerprinting,
+      ContentSettingsType::PLUGINS, brave_shields::kFingerprintingV2,
       CONTENT_SETTING_BLOCK);
   map->SetContentSettingCustomScope(
```
Comment (bridiver) [inline components/brave_shields/browser/brave_shields_util_unittest.cc:722]: this is not relevant anymore, there is no longer a first/third party scope. Can you please cleanup the tests in a followup?

Comment (simonhong) [reply components/brave_shields/browser/brave_shields_util_unittest.cc:722]: https://github.com/brave/brave-browser/issues/10078

**Pull request merged**